pypixelbuf.py

```diff
@@ -0,0 +1,389 @@
+# The MIT License (MIT)
+#
+# Based on the Adafruit NeoPixel and Adafruit Dotstar CircuitPython drivers.
+# Copyright (c) 2019 Roy Hooper
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+"""
+`pypixelbuf` - A pure python implementation of pixelbuf
+=======================================================
+This class is used when `pixelbuf` is not available in a build.
+
+This is a work in progress.
+
+* Author(s): Damien P. George &  Limor Fried & Scott Shawcroft & Roy Hooper
+"""
+import math
+
+
+class ByteOrder(object):
+    has_white = False
+    has_luminosity = False
+    bpp = 3
+    byteorder = (0, 1, 2)
+
+
+class RGB(ByteOrder):
+    byteorder = (0, 1, 2)
+
+
+class RBG(ByteOrder):
+    byteorder = (0, 2, 1)
+
+
+class GRB(ByteOrder):
+    byteorder = (1, 0, 2)
+
+
+class GBR(ByteOrder):
+    byteorder = (1, 2, 0)
+
+
+class BRG(ByteOrder):
+    byteorder = (2, 0, 1)
+
+
+class BGR(ByteOrder):
+    byteorder = (2, 1, 0)
+
+
+class RGBW(ByteOrder):
+    byteorder = (0, 1, 2, 3)
+    bpp = 4
+    has_white = True
+
+
+class RBGW(ByteOrder):
+    byteorder = (0, 2, 1, 3)
+    bpp = 4
+    has_white = True
+
+
+class GRBW(ByteOrder):
+    byteorder = (1, 0, 2, 3)
+    bpp = 4
+    has_white = True
+
+
+class GBRW(ByteOrder):
+    byteorder = (1, 2, 0, 3)
+    bpp = 4
+    has_white = True
+
+
+class BRGW(ByteOrder):
+    byteorder = (2, 0, 1, 3)
+    bpp = 4
+    has_white = True
+
+
+class BGRW(ByteOrder):
+    byteorder = (2, 1, 0, 3)
+    bpp = 4
+    has_white = True
+
+
+class LRGB(ByteOrder):
+    byteorder = (1, 2, 3, 0)
+    bpp = 4
+    has_luminosity = True
+
+
+class LRBG(ByteOrder):
+    byteorder = (1, 3, 2, 0)
+    bpp = 4
+    has_luminosity = True
+
+
+class LGRB(ByteOrder):
+    byteorder = (2, 1, 3, 0)
+    bpp = 4
+    has_luminosity = True
+
+
+class LGBR(ByteOrder):
+    byteorder = (2, 3, 1, 0)
+    bpp = 4
+    has_luminosity = True
```

Comment:
Weren't we going to replace these classes with strings like "RGB"?

Reply:
We are.  This was to get the PR in place to start moving forward.

Reply:
Ok, sounds good!